### PR TITLE
Add responsive props to Box

### PR DESCRIPTION
**Pull request by clxandstuff**
## Features
From now these props support responsive values:
- border
- noBorderRadius
- padding
- shadow

There were some more complex cases, when CSS selector depends on 2 responsive props. In order to implement that I created function to merge those props into one compatible with `generateResponsiveClassNames`

## Tests
- added `roots` to jest config to prevent running tests from build output directories like `esm`

### CODE REVIEW

File: src/components/utils/responsive-props.js [hunk 1]
```diff
@@ -10,6 +27,76 @@ export type ResponsivePropType<T> =
...
+          return acc;
+        },
+        {}
+      );
+    } else {
+      return {...prop};
+    }
+  });
+
+  // wyciągnij wspólne propsy dla breakpointów
```
Comment (coderitual) [inline src/components/utils/responsive-props.js:61]: eng please :)

Comment (clxandstuff) [reply src/components/utils/responsive-props.js]: Done

File: src/components/utils/responsive-props.js [hunk 2]
```diff
@@ -10,6 +27,76 @@ export type ResponsivePropType<T> =
...
+  const breakpoints = ['sm', 'md', 'lg', 'xl'];
+  let lastRowValues = [];
+
+  for (let i = 0; i <= 3; i++) {
```
[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/utils/responsive-props.js]: `forEach over breakpoints` the best 👍

**Review by coderitual — APPROVED** [review]
🚀

**Pull request merged**